### PR TITLE
Adding methods used for RHEL installations

### PR DESCRIPTION
**Pull request by vavuthu**
Signed-off-by: vavuthu <vavuthu@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -985,3 +985,17 @@ def get_operator_pods(operator_label=constants.OPERATOR_LABEL, namespace=None):
     operators = get_pods_having_label(operator_label, namespace)
     operator_pods = [Pod(**operator) for operator in operators]
     return operator_pods
+
+
+def upload_to_pod(pod_name, localpath, remotepath):
```
Comment (vasukulkarni) [inline ocs_ci/ocs/resources/pod.py]: s/to_pod/ is redundant,  just pod.upload() should be sufficient.

Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/templates/app-pods/rhel.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (petr-balogh) [inline ocs_ci/templates/app-pods/rhel.yaml]: Please name the file also with 7.7 like rhel-7.7.yaml

File: ocs_ci/ocs/constants.py [hunk 3]
```diff
@@ -256,6 +258,10 @@
     TEMPLATE_APP_POD_DIR, "fedora_dc.yaml"
 )
 
+RHEL_POD_YAML = os.path.join(
```
Comment (petr-balogh) [inline ocs_ci/ocs/constants.py]: Also include version

File: ocs_ci/ocs/resources/pod.py [hunk 4]
```diff
@@ -300,6 +300,55 @@ def run_git_clone(self):
...
+            packages (list): List of packages to install
+
+        """
+        if isinstance(packages, list):
+            packages = ' '.join(packages)
+
+        cmd = f"yum install {packages} -y"
+        self.exec_cmd_on_pod(cmd, False)
+
+    def copy_to_node(self, server, authkey, localpath, remotepath, user=None):
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/pod.py]: copy_to_server cause it's not the node yet when you call this function.

File: ocs_ci/ocs/resources/pod.py [hunk 5]
```diff
@@ -300,6 +300,55 @@ def run_git_clone(self):
...
+            authkey (str): Authentication file (.pem file)
+            localpath (str): Local file in pod to upload
+            remotepath (str): Target path on the remote server
+            user (str): User name to connect to server
+
+        """
+        if not user:
+            user = "root"
+
+        cmd = f"scp -i {authkey} {localpath} {user}@{server}:{remotepath}"
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/pod.py]: If we will add -r option it will allow us to copy also directories.

File: ocs_ci/ocs/resources/pod.py [hunk 6]
```diff
@@ -985,3 +1034,17 @@ def get_operator_pods(operator_label=constants.OPERATOR_LABEL, namespace=None):
...
+    """
+    Upload a file to pod
+
+    Args:
+        pod_name (str): Name of the pod
+        localpath (str): Local file to upload
+        remotepath (str): Target path on the pod
+
+    """
+    cmd = f"oc cp {os.path.expanduser(localpath)} {pod_name}:{remotepath}"
```
Comment (petr-balogh) [inline ocs_ci/ocs/resources/pod.py]: We can later on add this option (No need for this PR):  --no-preserve=false: The copied file/directory's ownership and permissions will not be preserved in the container

File: tests/helpers.py [hunk 7]
```diff
@@ -95,7 +95,7 @@ def wait_for_resource_state(resource, state, timeout=60):
 
 def create_pod(
     interface_type=None, pvc_name=None,
-    do_reload=True, namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+    do_reload=True, namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=None,
```
Comment (petr-balogh) [inline tests/helpers.py]: Put it as last one parameter as you are adding the new one.
You are missing it in docstring.

File: ocs_ci/templates/app-pods/rhel.yaml (hunk 2 above)
Comment (vavuthu) [reply ocs_ci/templates/app-pods/rhel.yaml]: Done

File: ocs_ci/ocs/constants.py (hunk 3 above)
Comment (vavuthu) [reply ocs_ci/ocs/constants.py]: Done

File: ocs_ci/ocs/resources/pod.py (hunk 4 above)
Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/ocs/resources/pod.py (hunk 5 above)
Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: If I use -r option , I am getting Identity issue . 
```
sh-4.2# scp -i /tmp/libra.pem /f1 test@rhel1-0.nov061.qe.rh-ocs.com:/tmp/
f1                                                                                                                                                          100%    0     0.0KB/s   00:00    
sh-4.2# 
sh-4.2# 
sh-4.2# scp -ir /tmp/libra.pem /f1 test@rhel1-0.nov061.qe.rh-ocs.com:/tmp/
Warning: Identity file r not accessible: No such file or directory.
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
lost connection
sh-4.2# scp -ir /tmp/libra.pem /test test@rhel1-0.nov061.qe.rh-ocs.com:/tmp/
Warning: Identity file r not accessible: No such file or directory.
Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
lost connection
sh-4.2# 
```

I will debug later and take it in next PR

File: ocs_ci/ocs/resources/pod.py (hunk 6 above)
Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Agreed. I will take it up as followup PR

File: tests/helpers.py (hunk 7 above)
Comment (vavuthu) [reply tests/helpers.py]: Done

File: ocs_ci/ocs/resources/pod.py (hunk 5 above)
Comment (petr-balogh) [reply ocs_ci/ocs/resources/pod.py]: after -i you need to have key so that's the issue here.  So you need to make it like
`scp -i /tmp/libra.pem -r ...`

Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/ocs/resources/pod.py [hunk 8]
```diff
@@ -300,6 +300,58 @@ def run_git_clone(self):
...
+
+        Args:
+            packages (list): List of packages to install
+
+        """
+        if isinstance(packages, list):
+            packages = ' '.join(packages)
+
+        cmd = f"yum install {packages} -y"
+        self.exec_cmd_on_pod(cmd, False)
```
Comment (dahorak) [inline ocs_ci/ocs/resources/pod.py]: I would suggest to add the parameter name for the `False` parameter, for better readability, without checking the method definition:
```suggestion
        self.exec_cmd_on_pod(cmd, exec_cmd_on_pod=False)
```
(similarly on other places below)

File: ocs_ci/ocs/resources/pod.py [hunk 9]
```diff
@@ -300,6 +300,58 @@ def run_git_clone(self):
...
+
+        """
+        if not user:
+            user = "root"
+
+        cmd = (
+            f"scp -i {authkey} -o \"StrictHostKeyChecking no\""
+            f" -r {localpath} {user}@{server}:{remotepath}"
+        )
+        self.exec_cmd_on_pod(cmd, False)
```
Comment (dahorak) [inline ocs_ci/ocs/resources/pod.py]:
```suggestion
        self.exec_cmd_on_pod(cmd, exec_cmd_on_pod=False)
```

File: ocs_ci/ocs/resources/pod.py [hunk 10]
```diff
@@ -300,6 +300,58 @@ def run_git_clone(self):
...
+            authkey (str): Authentication file (.pem file)
+            cmd (str): command to run on server from pod
+            user (str): User name to connect to server
+
+        """
+        if not user:
+            user = "root"
+
+        cmd = f"ssh -i {authkey} -o \"StrictHostKeyChecking no\" {user}@{server} {cmd}"
+        self.exec_cmd_on_pod(cmd, False)
```
Comment (dahorak) [inline ocs_ci/ocs/resources/pod.py]:
```suggestion
        self.exec_cmd_on_pod(cmd, exec_cmd_on_pod=False)
```

File: ocs_ci/templates/ocp-deployment/inventory.yaml.j2 [hunk 11]
```diff
@@ -0,0 +1,11 @@
+[all:vars]
+ansible_user={{ ansible_user | default('test') }}
+ansible_become={{ ansible_become | default('True') }}
+
+openshift_kubeconfig_path={{ pod_kubeconfig }}
+openshift_pull_secret_path={{ pod_pull_secret | default('True') }}
```
Comment (dahorak) [inline ocs_ci/templates/ocp-deployment/inventory.yaml.j2]: Is default value `True` correct for `..._path` variable?

File: ocs_ci/ocs/resources/pod.py (hunk 8 above)
Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/ocs/resources/pod.py (hunk 9 above)
Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/ocs/resources/pod.py (hunk 10 above)
Comment (vavuthu) [reply ocs_ci/ocs/resources/pod.py]: Done

File: ocs_ci/templates/ocp-deployment/inventory.yaml.j2 (hunk 11 above)
Comment (vavuthu) [reply ocs_ci/templates/ocp-deployment/inventory.yaml.j2]: Done

**Pull request merged**